enterprise_access/apps/bffs/handlers.py

```diff
@@ -363,13 +390,20 @@ def check_and_auto_apply_license(self):
         """
         Check if auto-apply licenses are available and apply them to the user.
         """
-        subscription_licenses_by_status = self.subscription_licenses_by_status
-        assigned_licenses = subscription_licenses_by_status.get('assigned', [])
-
-        if assigned_licenses or self.current_active_license:
-            # Skip auto-apply if user already has assigned license(s) or an already-activated license
+        if (
+            self.current_assigned_licenses or
+            self.current_activated_licenses or
+            self.current_revoked_licenses or
```

Comment:
Maybe we can add a helper to produce a list of all current licenses, regardless of status, and then use that here?

Reply:
Ah, good suggestion given all 3 allocatable license statuses are considered now; we can use the existing `self.subscription_licenses` 😄